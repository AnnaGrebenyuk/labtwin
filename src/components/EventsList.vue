<template>
  <b-container>
      <b-list-group>
        <b-list-group-item cols="3" v-for="event in events" :key="event.id">
          <events-list-item :event="event" :favoritesMode="favoritesMode"/>
        </b-list-group-item>
    </b-list-group>
  </b-container>
</template>

<script>
import { BContainer, BListGroup, BListGroupItem } from 'bootstrap-vue';
import EventsListItem from './EventsListItem.vue';

export default {
  name: 'EventsList',
  components: {
    EventsListItem, BContainer, BListGroup, BListGroupItem,
  },
  props: {
    events: Array,
    favoritesMode: Boolean,
  },
};
</script>
