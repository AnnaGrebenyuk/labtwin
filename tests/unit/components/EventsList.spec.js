import {
  shallowMount,
  createLocalVue,
} from '@vue/test-utils';
import Vuex from 'vuex';
import EventsList from '@/components/EventsList.vue';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('EventsList', () => {
  it('renders correctly', () => {
    const wrapper = shallowMount(EventsList);
    expect(wrapper.element).toMatchSnapshot();
  });
});
