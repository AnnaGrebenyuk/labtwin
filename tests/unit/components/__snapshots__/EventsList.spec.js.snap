// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventsList renders props.event.name when passed 1`] = `
<b-container-stub
  tag="div"
>
  <b-list-group-stub
    tag="div"
  />
</b-container-stub>
`;
